nemo/collections/nlp/modules/common/megatron/adapters/mcore_mixins.py

```diff
@@ -111,6 +117,17 @@ def get_query_key_value_tensors(self, hidden_states, key_value_states=None):
 
                 mixed_qkv = mixed_qkv + lora_mixed_qkv
 
+            # (@adithyare) TODO: this setup does not check if both fused and unfused adapters are enabled.
+            # Should probably add this check.
+            lora_unfused_kqv_adapter = self.get_adapter_module(AdapterName.LORA_UNFUSED_KQV_ADAPTER)
+            if lora_unfused_kqv_adapter and self.adapter_cfg[AdapterName.LORA_UNFUSED_KQV_ADAPTER]['enabled']:
+                if layernorm_output is not None:
+                    lora_mixed_qkv = lora_unfused_kqv_adapter(layernorm_output)
+                else:
+                    lora_mixed_qkv = lora_unfused_kqv_adapter(hidden_states)
+
+                mixed_qkv = mixed_qkv + lora_mixed_qkv
+
```

Comment:
i wonder if you can reuse lines 107-112 (in original file) here? so maybe check which lora is available, and only have the variable `lora_kqv_adapter` point to the corresponding adapter module